modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -5088,14 +5106,16 @@ private void unlockListenerReadLock() {
 
     /** {@inheritDoc} */
     @Override public String toString() {
-        lockEntry();
-
-        try {
-            return S.toString(GridCacheMapEntry.class, this);
-        }
-        finally {
-            unlockEntry();
+        if (tryLockEntry(ENTRY_LOCK_TIMEOUT)) {
```

Comment:
This piece of code is repeated three times, we can extract it to the eldest class (GridCacheMapEntry).
I believe, there's no need to use this logic in GridCacheTestEntryEx.